src/lib/full_upgrade.sh

```diff
@@ -50,7 +50,7 @@ source "${libdir}/packages_cache.sh"
 source "${libdir}/pacnew_files.sh"
 
 # Source the "kernel_reboot" library which checks if there's a pending kernel update requiring a reboot to be applied (unless running from a container)
-if ! systemd-detect-virt --container --quiet; then
+if ! pacman -Qi kernel-modules-hook &>/dev/null && ! systemd-detect-virt --container --quiet; then
```

Comment:
```suggestion
if ! pacman -Q kernel-modules-hook &> /dev/null || ! systemd-detect-virt --container --quiet; then
```

- We don't need to print the package's info with `-Qi` here (the output is redirected to `/dev/null` anyway :stuck_out_tongue:), `-Q` is enough.
- We need a logical "OR", not a logical "AND" (as in, "source the `kernel_reboot.sh` script if the `kernel-modules-hook` package is not installed OR we are not running from a container").

Reply:
Updated. I had missed that the comment above it mentioned the existing exception, which is why I hadn't adjusted it. All addressed now.